src/pke/lib/scheme/ckksrns/ckksrns-pke.cpp

```diff
@@ -0,0 +1,95 @@
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//==================================================================================
+
+/*
+CKKS implementation. If NOISE_FLOODING_DECRYPT is set, we flood the decryption bits with noise.
+ */
+
+#define PROFILE
+
+#include "cryptocontext.h"
+#include "scheme/ckksrns/ckksrns-cryptoparameters.h"
+#include "scheme/ckksrns/ckksrns-pke.h"
+
+namespace lbcrypto {
+
+DecryptResult PKECKKSRNS::Decrypt(ConstCiphertext<DCRTPoly> ciphertext, const PrivateKey<DCRTPoly> privateKey,
+                                  NativePoly* plaintext) const {
+    const auto cryptoParams = std::dynamic_pointer_cast<CryptoParametersCKKSRNS>(ciphertext->GetCryptoParameters());
+    const std::vector<DCRTPoly>& cv = ciphertext->GetElements();
+    DCRTPoly b                      = DecryptCore(cv, privateKey);
+    if (cryptoParams->GetDecryptionNoiseMode() == NOISE_FLOODING_DECRYPT &&
+        cryptoParams->GetExecutionMode() == EXEC_EVALUATION) {
+        auto dgg = cryptoParams->GetFloodingDiscreteGaussianGenerator();
+        DCRTPoly noise(dgg, cv[0].GetParams(), Format::EVALUATION);
+        b += noise;
+    }
+
+    b.SetFormat(Format::COEFFICIENT);
+    const size_t sizeQl = b.GetParams()->GetParams().size();
+
+    if (sizeQl == 1)
+        *plaintext = b.GetElementAtIndex(0);
+    else
+        OPENFHE_THROW(math_error, "Decryption failure: No towers left; consider increasing the depth.");
+
+    return DecryptResult(plaintext->GetLength());
+}
+
+DecryptResult PKECKKSRNS::Decrypt(ConstCiphertext<DCRTPoly> ciphertext, const PrivateKey<DCRTPoly> privateKey,
+                                  Poly* plaintext) const {
+    const auto cryptoParams = std::dynamic_pointer_cast<CryptoParametersCKKSRNS>(ciphertext->GetCryptoParameters());
+    const std::vector<DCRTPoly>& cv = ciphertext->GetElements();
+    DCRTPoly b                      = DecryptCore(cv, privateKey);
+    if (cryptoParams->GetDecryptionNoiseMode() == NOISE_FLOODING_DECRYPT &&
+        cryptoParams->GetExecutionMode() == EXEC_EVALUATION) {
+        auto dgg = cryptoParams->GetFloodingDiscreteGaussianGenerator();
+        DCRTPoly noise(dgg, cv[0].GetParams(), Format::EVALUATION);
+        b += noise;
+    }
+
+    b.SetFormat(Format::COEFFICIENT);
+    const size_t sizeQl = b.GetParams()->GetParams().size();
+
+    if (sizeQl > 1) {
```

Comment:
i would recommend to have if(sizeQl == 0) first (to throw an exception) and after that :
if(sizeQl == 1) {
.....
}
else {
.....
}

Reply:
Added if, then if-else.